models/parrot_rate_neuron.h

```diff
@@ -0,0 +1,287 @@
+/*
+ *  parrot_rate_neuron.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef PARROT_RATE_NEURON_H
+#define PARROT_RATE_NEURON_H
+
+#include "config.h"
+
+#include "archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "node.h"
+#include "normal_randomdev.h"
+#include "poisson_randomdev.h"
+#include "ring_buffer.h"
+#include "recordables_map.h"
+#include "universal_data_logger.h"
+
+#include <string>
+
+namespace nest
+{
+
+/* BeginDocumentation
+Name: parrot_rate_neuron - Rate neuron that sums up incoming rates and applies
+                           a nonlinearity specified via the template.
+
+Description:
+
+The parrot rate neuron simply sums up all incoming rates and applies
+the nonlinearity specified in the function input of the template class.
+An important application is to provide the possibility to
+apply different nonlinearities to different incoming connections of the
+same rate neuron by connecting the sending rate neurons to the
+parrot rate neuron and connecting the parrot rate neuron to the
+receiving rate neuron instead of using a direct connection.
+Please note that for instantaneous rate connections the rate arrives
+one time step later at the receiving rate neurons as with a direct connection.
+
+Remarks:
+
+- Weights on connections from and to the parrot_rate_neuron
+  are handled as usual.
+- Delays are honored on incoming and outgoing connections.
+
+Receives: InstantaneousRateConnectionEvent, DelayedRateConnectionEvent
+
+Sends: InstantaneousRateConnectionEvent, DelayedRateConnectionEvent
+
+Parameters:
+No parameters to be set in the status dictionary.
+
+Author: Mario Senden, Jan Hahne, Jannis Schuecker
+FirstVersion: November 2017
+*/
+template < class TNonlinearities >
```

Comment:
`class` seems to be used in many templates at the moment (i looked for example in rate_neuron_ipn/opn and the binary neuron). Maybe this can rather be fixed in an separate PR for all NEST files?